src/Seascope.py

```diff
@@ -395,7 +395,12 @@ def closeEvent(self, ev):
 
 		self.app_write_config()
 		if backend.proj_is_open():
-			self.app_gui_state_save(backend.proj_dir())
+			if DialogManager.show_yes_no('Save session?'):
+				self.app_gui_state_save(backend.proj_dir())
+			else:
+				f = self.app_gui_state_file()
```

Comment:
Deleting the existing session file, which might be a session prior to the current one, appears to be too aggressive and unexpected?

I'd suggest do nothing if the user choose not to save session, which would keep previously saved state.

Reply:
I agree it might be too aggressive. I chose now to use a custom dialog presenting the user all three options. Please find it in my updated branch (now also rebased to current master). Thank you for the review and merging my other PRs!